packages/app/src/app/overmind/namespaces/git/actions.ts

```diff
@@ -812,6 +812,11 @@ export const _compareWithSource: AsyncAction = async ({
     true
   );
 
+  if (!originalChanges) {
```

Comment:
I would rather put a `try/catch` around the call above to `compareGit`. That way the reason for it being there is closer to the modal logic and you do not have to deal with pesky "possibly null"

```ts
try {
const originalChanges = await effects.api.compareGit(
    sandbox.id,
    sandbox.originalGitCommitSha!,
    sandbox.originalGit!.branch,
     true
   );
   const updates = await actions.git._evaluateGitChanges(baseChanges.files);
   // and so on
} catch (error) {
  state.currentModal = 'notFoundBranchModal';
}
```

Reply:
That still showed the error that's why I did it this way

Reply:
Though did that error come from this? Or is there maybe some more logic running creating the error? What does the stack trace say? 😄

Reply:
Should return from the `catch` so no other logic keeps running. Gimme call tomorrow and we can take a look together 😄

Reply:
I'll call you arround eleven to explain better :)